app-frontend/src/app/pages/library/projects/detail/projectScenes/projectScenes.controller.js

```diff
@@ -186,4 +189,37 @@ export default class ProjectScenesController {
             }
         );
     }
-}
+
+    toggleProjectNameEdit() {
+        if (this.isEditingProjectName) {
+            this.cancelProjectNameEdit();
+        } else {
+            this.startProjectNameEdit();
+        }
+    }
+
+    startProjectNameEdit() {
+        this.editedProjectName = this.project.name;
+        this.isEditingProjectName = true;
+    }
+
+    saveProjectNameEdit() {
+        this.isSavingProjectNameEdit = true;
+        this.isEditingProjectName = false;
+        this.project.name = this.editedProjectName;
+        this.projectService.updateProject(this.project).then(
+            () => {
+                this.isSavingProjectNameEdit = false;
+            },
+            (err) => {
+                this.$log.error('Error renaming project:', err);
```

Comment:
If there's an error, we shouldn't show the project name as changed. Also, since we track the state of the request, we can have a spinner that shows that the action is in progress (maybe replace the two buttons with it).